direct/nn/vsharp/vsharp_engine.py

```diff
@@ -18,6 +18,134 @@
 from direct.utils import detach_dict, dict_to_device
 
 
+class VSharpNet3DEngine(MRIModelEngine):
+    """VSharpNet Engine."""
+
+    def __init__(
+        self,
+        cfg: BaseConfig,
+        model: nn.Module,
+        device: str,
+        forward_operator: Optional[callable] = None,
+        backward_operator: Optional[callable] = None,
```

Comment:
I think you can do callable[[torch.Tensor], torch.Tensor] or something like that. You can also replace the optional with the Union operator with None, which is a shortcut for ` | None`

Reply:
Same as above. I think Optional is more clean than | None.